Modules/Sources/AsyncImageKit/Views/CachedAsyncImage.swift

```diff
@@ -1,25 +1,33 @@
 import SwiftUI
 
+public protocol ImageURLResolver: Sendable, Identifiable {
```

Comment:
It seems like a bit of an over-kill to add a protocol for this one use-case.

I'd suggest using composition:

```swift
struct PluginIconView: View {
    let slug: String
    @State var iconURL: URL?

    var body: some View {
        ZStack {
            if let iconURL {
                CachedAsyncImage(url: iconURL)
                /// ...
            } else {
                /// loading state
            }
        }
        .task {
            iconURL = try? await resolveIconURL(for: slug)
        }
    }
}
```

It's less code, and it won't require adding complexity to `CachedAsyncImage`.

Reply:
Oh nice. I'll use your suggested approach instead.